models/models.go

```diff
@@ -12,10 +13,32 @@ import (
 	"github.com/18F/cf-cdn-service-broker/utils"
 )
 
+type State string
+
+const (
+	Provisioning   State = "provisioning"
+	Provisioned          = "provisioned"
+	Deprovisioning       = "deprovisioning"
+	Deprovisioned        = "deprovisioned"
+)
+
+func (s State) Value() (driver.Value, error) {
+	return string(s), nil
+}
+
+func (s *State) Scan(value interface{}) error {
+	bytes, ok := value.([]byte)
+	if !ok {
+		return fmt.Errorf("error scanning status %s", value)
+	}
+	*s = State(bytes)
+	return nil
+}
```

Comment:
Yep, this is the gross stuff that I was avoiding before to get custom types to work here. The orm and database driver know how to marshal and unmarshal certain types to and from the db, like `string` and `int64`, but they don't know what to do with custom types. These methods make our `State` type satisfy the `Valuer` and `Scanner` interfaces, which the database driver will use if available on converting go values to the database and vice versa. The `Value` method converts a `State` to a `string`, and the `Scan` method converts what we hope is a `[]byte` (otherwise we'll error) to a `State`.